netutils/netcat/netcat_main.c

```diff
@@ -44,18 +44,23 @@
 # define NETCAT_PORT 31337
 #endif
 
+#ifndef NETCAT_IOBUF_SIZE
+# define NETCAT_IOBUF_SIZE 256
+#endif
+
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
 int do_io(int infd, int outfd)
 {
-  size_t capacity = 256;
-  char buf[capacity];
+  ssize_t avail;
+  ssize_t written;
+  char buf[NETCAT_IOBUF_SIZE];
 
   while (true)
     {
-      ssize_t avail = read(infd, buf, capacity);
```

Comment:
Actually this is not a C89 issue. Standard allows this.

Reply:
Yes, "ssize_t avail" was moved not because of C89. I moved it together with moving "ssize_t written" that was C89 related issue. So I moved "ssize_t avail" for uniformity reasons.